local/local.go

```diff
@@ -310,16 +269,19 @@ func (i *Image) TopLayer() (string, error) {
 }
 
 func (i *Image) GetLayer(diffID string) (io.ReadCloser, error) {
-	err := i.downloadImageOnce(i.repoName)
-	if err != nil {
-		return nil, err
+	for l := range i.inspect.RootFS.Layers {
+		if i.inspect.RootFS.Layers[l] != diffID {
+			continue
+		}
+		if i.layerPaths[l] == "" {
+			if err := i.downloadBaseLayersOnce(); err != nil {
+				return nil, err
+			}
+		}
+		return os.Open(i.layerPaths[l])
```

Comment:
An edge case for sure but wouldn't it be better to check if `i.layerPaths[l]` is valid/non-empty after downloading the base layers? Otherwise, a very cryptic error may be thrown if that edge case is ever encountered.